plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1492,6 +1501,7 @@ private void finishWrite(ConnectorSession session, IcebergTableHandle table, Col
 
             rowDelta.validateDataFilesExist(referencedDataFiles.build());
             try {
+                getExtraSummaryMetadata(session).forEach(rowDelta::set);
```

Comment:
does the setting of extra properties need to be inside `try-catch` ?

Reply:
No it doesn't.

Reply:
Moved it out of that block =)